src/liballoc/arc.rs

```diff
@@ -282,6 +282,9 @@ impl<T> Arc<T> {
     /// To avoid a memory leak the pointer must be converted back to an `Arc` using
     /// [`Arc::from_raw`][from_raw].
     ///
+    /// If `T` is zero-sized (e.g. `Arc<()>`), the returned pointer address
+    /// will be meaningless.
```

Comment:
I think for `Rc` and `Arc` this is wrong, because `from_raw` exists and there's the two refcounts, not just the ZST.